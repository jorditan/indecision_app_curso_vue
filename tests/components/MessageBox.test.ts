import MessageBox from '@/components/chat/MessageBox.vue';
import { mount } from '@vue/test-utils';

const wrapper = mount(MessageBox);

describe('<MessageBox/>', () => {
  test('Rendeser input and button elements correctly', () => {
    expect(wrapper.html).toMatchSnapshot();
    expect(wrapper.find('input[type="text"]').exists()).toBe(true);
    expect(wrapper.find('button').exists()).toBe(true);
    expect(wrapper.find('button svg').exists()).toBe(true);
  });

  test('Emits sendMessage event when the button is clicked with message value', async () => {
    const message = 'Hola mundo';
    await wrapper.find('input[type=text]').setValue(message);
    await wrapper.find('button').trigger('click');

    expect(wrapper.emitted('sendMessage')?.[0]).toEqual([message]);
    expect((wrapper.vm as any).message).toBe('');
  });

  test('Emits sendMessage event when keypress is triggered', async () => {
    const message = 'Hola mundo';
    const input = wrapper.find('input');

    await input.setValue(message);
    await wrapper.find('input[type=text]').setValue(message);
    await input.trigger('keypress.enter');
    await wrapper.find('button').trigger('click');

    expect(wrapper.emitted('sendMessage')?.[0]).toEqual([message]);
    expect((wrapper.vm as any).message).toBe('');
  });

  test('Emits sendMessage event when keypress is triggered is clicked with message value', async () => {
    const wrapper = mount(MessageBox);
    const input = wrapper.find('input');

    await input.trigger('keypress.enter');
    await wrapper.find('button').trigger('click');

    expect(wrapper.emitted('sendMessage')).toBeFalsy();
  });
});
