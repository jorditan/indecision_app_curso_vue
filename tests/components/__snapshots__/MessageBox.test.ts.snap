// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MessageBox/> > Rendeser input and button elements correctly 1`] = `[Function]`;
